### PR TITLE
Setting clearance in mobileui

### PR DESCRIPTION
**Pull request by siemiatj**

### CODE REVIEW

**Comment by siemiatj** [comment]
@teosarca once the status is selected, `allowedClearanceStatuses` returns a list without this status. So when user changes it, we should re-fetch the statuses ?

rENfqoBxhy


https://recordit.co/rENfqoBxhy

**Comment by teosarca** [comment]
@siemiatj  don't u have the current status in handlingUnits info?

File: misc/services/mobile-webui/mobile-webui-frontend/src/apps/huManager/components/ClearanceDialog.jsx [hunk 1]
```diff
@@ -63,10 +63,10 @@ const ClearanceDialog = ({ handlingUnitInfo, clearanceStatuses, onClearanceChang
               </tbody>
             </table>
             <div className="buttons is-centered">
-              <button className="button is-danger" onClick={onDialogYes}>
+              <button className="button is-success" onClick={onDialogYes}>
```
Comment (teosarca) [inline misc/services/mobile-webui/mobile-webui-frontend/src/apps/huManager/components/ClearanceDialog.jsx:63]: the OK button shall be danger.... but it's actually changing things.

File: misc/services/mobile-webui/mobile-webui-frontend/src/apps/huManager/api.js [hunk 2]
```diff
@@ -36,3 +36,25 @@ export const moveHU = ({ huId, huQRCode, targetQRCode }) => {
...
+    .then(unboxAxiosResponse)
+    .then((response) => response.clearanceStatuses);
+}
+
+export function setClearanceStatusRequest({ huId, clearanceNote, clearanceStatus }) {
+  const dataObject = {};
+  if (clearanceNote) {
+    dataObject.clearanceNote = clearanceNote;
+  }
+  if (clearanceStatus) {
```
Comment (teosarca) [inline misc/services/mobile-webui/mobile-webui-frontend/src/apps/huManager/api.js:55]: what's the reason for not having a clearanceStatus?

File: misc/services/mobile-webui/mobile-webui-frontend/src/apps/huManager/api.js [hunk 3]
```diff
@@ -36,3 +36,25 @@ export const moveHU = ({ huId, huQRCode, targetQRCode }) => {
...
+export function getAllowedClearanceStatusesRequest({ huId }) {
+  return axios
+    .get(`${huAPIBasePath}/byId/${huId}/allowedClearanceStatuses`)
+    .then(unboxAxiosResponse)
+    .then((response) => response.clearanceStatuses);
+}
+
+export function setClearanceStatusRequest({ huId, clearanceNote, clearanceStatus }) {
+  const dataObject = {};
+  if (clearanceNote) {
```
Comment (teosarca) [inline misc/services/mobile-webui/mobile-webui-frontend/src/apps/huManager/api.js:52]: IMHO in this case u can send null.

File: misc/services/mobile-webui/mobile-webui-frontend/src/apps/huManager/components/HUInfoComponent.jsx [hunk 4]
```diff
@@ -19,6 +20,10 @@ export const HUInfoComponent = ({ handlingUnitInfo }) => {
           <th>{trl('huManager.locator')}</th>
           <td>{handlingUnitInfo.locatorValue}</td>
         </tr>
+        <tr>
+          <th>{trl('huManager.clearanceStatus')}</th>
+          <td>{clearanceStatus}</td>
+        </tr>
```
Comment (teosarca) [inline misc/services/mobile-webui/mobile-webui-frontend/src/apps/huManager/components/HUInfoComponent.jsx:26]: Q: don't we need to show the clearance note if any?

**Review by teosarca — CHANGES_REQUESTED** [review]
check my comments

File: misc/services/mobile-webui/mobile-webui-frontend/src/apps/huManager/actionTypes.js [hunk 5]
```diff
@@ -1,2 +1,3 @@
 export const CLEAR_LOADED_DATA = 'huManager/CLEAR_LOADED_DATA';
 export const HANDLING_UNIT_LOADED = 'huManager/HANDLING_UNIT_LOADED';
+export const CHANGE_CLEARANCE_STATUS = 'hiManager/CHANGE_CLEARANCE_STATUS';
```
Comment (teosarca) [inline misc/services/mobile-webui/mobile-webui-frontend/src/apps/huManager/actionTypes.js:3]: `huManager` (not hi...)

File: misc/services/mobile-webui/mobile-webui-frontend/src/apps/huManager/api.js (hunk 2 above)
Comment (teosarca) [reply misc/services/mobile-webui/mobile-webui-frontend/src/apps/huManager/api.js:53]: > If it didn't change or user only added clearanceNote, should we send it anyway ?

Nop, a clearanceNote makes no sense without a status.
The `note` is the explanation of the `status`.

e.g. Status:Quarantine, Note: A bird took a dump on it

File: misc/services/mobile-webui/mobile-webui-frontend/src/apps/huManager/i18n/de.json [hunk 6]
```diff
@@ -3,6 +3,8 @@
   "qrCode": "QR code",
   "locator": "Locator",
   "product": "Product",
+  "clearanceStatus": "Clearance Status",
```
Comment (teosarca) [inline misc/services/mobile-webui/mobile-webui-frontend/src/apps/huManager/i18n/de.json:6]: we need trl for this too

**Comment by siemiatj** [comment]
aYULIb75Wh

**Comment by teosarca** [comment]
@siemiatj  one thing I remarked on ur screenshot: pls keep the `Scan again` button last.

**Pull request merged**